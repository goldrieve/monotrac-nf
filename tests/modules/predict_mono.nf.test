nextflow_process {

    name "Test Process PREDICT"
    script "main.nf"
    process "PREDICT"

    test("Should predict outbreak potential - monomorphic") {

        when {
            params {
                pkl = "${projectDir}/data/machine_learning/ml.pkl"
            }
            process {
                """
                input[0] = ["SVP_test", file("${projectDir}/tests/data/SVP_test_AA_counts.csv")]
                input[1] = params.pkl
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}