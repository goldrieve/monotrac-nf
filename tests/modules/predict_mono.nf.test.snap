{
    "Should predict outbreak potential - monomorphic": {
        "content": [
            {
                "0": [
                    "SVP_test_prediction.txt:md5,f67624311864b033b2b3e079c3733901"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T14:25:10.571331841"
    }
}